src/shared/networkrequest.cpp

```diff
@@ -0,0 +1,487 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "networkrequest.h"
+
+#include <QDirIterator>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QNetworkRequest>
+#include <QRegularExpression>
+#include <QUrl>
+
+#include "leakdetector.h"
+#include "logger.h"
+#include "networkmanager.h"
+#include "settingsholder.h"
+#include "task.h"
+
+// Timeout for the network requests.
+constexpr uint32_t REQUEST_TIMEOUT_MSEC = 15000;
```

Comment:
Is this enough time for the delete devices request?

Reply:
I see we use disableTimeout for add device and balrog.

Reply:
Right. Let's do it here as well.